ipn/ipnlocal/local.go

```diff
@@ -4624,6 +4632,15 @@ func (b *LocalBackend) applyPrefsToHostinfoLocked(hi *tailcfg.Hostinfo, prefs ip
 	hi.ShieldsUp = prefs.ShieldsUp()
 	hi.AllowsUpdate = envknob.AllowsRemoteUpdate() || prefs.AutoUpdate().Apply.EqualBool(true)
 
+	// count routes without exit node routes
+	routeCount := expvar.Int{}
+	for _, route := range hi.RoutableIPs {
+		if route.Bits() != 0 {
+			routeCount.Add(1)
+		}
+	}
+	metricAdvertisedRoutes.Set(struct{}{}, &routeCount)
```

Comment:
`Set` creates a new metric every time, and I don't think it's supposed to be used like this.

Actually, I am not sure what it's supposed to be used for, since it's not currently used. Perhaps we can change it to behave in a similar way to [Add](https://github.com/tailscale/tailscale/blob/f23932bd9858539ce4730b413709f4febb5bac93/metrics/multilabelmap.go#L215)?

I'd perhaps change `Set`

Reply:
`Set` is used in control, and useful when using `expvar.Func`, so I opted for adding `SetInt` and `SetFloat` at this time.